### PR TITLE
mcs: support scheduler config forward and enable some tests

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5839

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: tests/pdctl/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -160,8 +169,11 @@ func (suite *schedulerTestSuite) checkScheduler(cluster *tests.TestCluster) {
 		checkSchedulerCommand(args, expected)
 
 		// check update success
-		expectedConfig["store-id-ranges"] = map[string]interface{}{"2": []interface{}{map[string]interface{}{"end-key": "", "start-key": ""}}, "3": []interface{}{map[string]interface{}{"end-key": "", "start-key": ""}}}
-		checkSchedulerConfigCommand(expectedConfig, schedulers[idx])
+		// FIXME: remove this check after scheduler config is updated
```
Comment (lhy1024) [inline tests/pdctl/scheduler/scheduler_test.go:172]: But scheduling server cannot update config

File: pkg/schedule/handler/handler.go [hunk 2]
```diff
@@ -837,7 +837,19 @@ func (h *Handler) GetSchedulerByStatus(status string, needTS bool) (interface{},
 		}
 		return disabledSchedulers, nil
 	default:
-		return schedulers, nil
+		// The default scheduler could not be deleted, it could only be disabled.
+		// TODO: Should we distinguish between disabled and removed schedulers?
+		var enabledSchedulers []string
```
Comment (lhy1024) [inline pkg/schedule/handler/handler.go:842]: Should we distinguish between disabled and removed schedulers?

Comment (lhy1024) [reply pkg/schedule/handler/handler.go:842]: After discussion, we will not distinguish them.

File: tests/pdctl/scheduler/scheduler_test.go (hunk 1 above)
Comment (lhy1024) [reply tests/pdctl/scheduler/scheduler_test.go:172]: https://github.com/tikv/pd/issues/7257

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 3]
```diff
@@ -128,6 +129,7 @@ func (s *Service) RegisterSchedulersRouter() {
 	router := s.root.Group("schedulers")
 	router.GET("", getSchedulers)
 	router.GET("/diagnostic/:name", getDiagnosticResult)
+	router.GET("/config/:name/:suffix", getSchedulerConfigByName)
 	// TODO: in the future, we should split pauseOrResumeScheduler to two different APIs.
 	// And we need to do one-to-two forwarding in the API middleware.
 	router.POST("/:name", pauseOrResumeScheduler)
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:135]: How about unifying these formats?

Comment (lhy1024) [reply pkg/mcs/scheduling/server/apis/v1/api.go:135]: How about 
```
router.GET("", getSchedulers)
router.GET("/:name", getDiagnosticResult)
router.POST("/:name", pauseOrResumeScheduler)
router.GET("/config/:name/:suffix", getSchedulerConfigByName)
```

Comment (rleungx) [reply pkg/mcs/scheduling/server/apis/v1/api.go:135]: How about using 
router.GET("", getSchedulers)
router.GET("/:name/diagnose", getDiagnosticResult)
router.POST("/:name", pauseOrResumeScheduler)
router.GET("/:name/config", getSchedulerConfigByName)

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 4]
```diff
@@ -128,6 +131,9 @@ func (s *Service) RegisterSchedulersRouter() {
 	router := s.root.Group("schedulers")
 	router.GET("", getSchedulers)
 	router.GET("/diagnostic/:name", getDiagnosticResult)
+	router.GET("/config", getSchedulerConfig)
+	router.GET("/config/:name/list", getSchedulerConfigByName)
+	router.GET("/config/:name/roles", getSchedulerConfigByName) // compatibility for shuffle-region-scheduler
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:136]: Do we need these two?

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 5]
```diff
@@ -385,6 +391,58 @@ func getSchedulers(c *gin.Context) {
 	c.IndentedJSON(http.StatusOK, output)
 }
 
+// @Tags     schedulers
+// @Summary  List all scheduler configs.
+// @Produce  json
+// @Success  200  {object}  map[string]interface{}
```
Comment (rleungx) [inline pkg/mcs/scheduling/server/apis/v1/api.go:397]: need to update

File: pkg/mcs/scheduling/server/apis/v1/api.go (hunk 4 above)
Comment (lhy1024) [reply pkg/mcs/scheduling/server/apis/v1/api.go:136]: Yes, otherwise we'll modify shuffle region scheduler

Comment (lhy1024) [reply pkg/mcs/scheduling/server/apis/v1/api.go:136]: Or we may add a new interface for it

Comment (lhy1024) [reply pkg/mcs/scheduling/server/apis/v1/api.go]: unify to "/list" and use new api in pd-ctl

**Comment by lhy1024** [comment]
@rleungx @jmpotato ptal

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**